packages/fuels-accounts/src/wallet.rs

```diff
@@ -303,14 +309,13 @@ impl Signer for WalletUnlocked {
     }
 
     fn sign_transaction(&self, tx: &mut impl Transaction) -> WalletResult<Signature> {
-        let id = tx.id();
-
-        // Safety: `Message::from_bytes_unchecked` is unsafe because
-        // it can't guarantee that the provided bytes will be the product
-        // of a cryptographically secure hash. However, the bytes are
-        // coming from `tx.id()`, which already uses `Hasher::hash()`
-        // to hash it using a secure hash mechanism.
-        let message = unsafe { Message::from_bytes_unchecked(*id) };
+        let consensus_parameters = self
+            .try_provider()
+            .map_err(|_| WalletError::NoProviderError)?
+            .consensus_parameters();
```

Comment:
This is what I was referring to above regarding signing without a provider. Here we could unwrap or use the default value.